plugins/polio/js/src/domains/Campaigns/Scope/ScopeForm.tsx

```diff
@@ -77,44 +77,47 @@ export const ScopeForm: FunctionComponent = () => {
     }, [currentTab, rounds, scopePerRound, sortedRounds, values.scopes]);
 
     const filteredDistricts: FilteredDistricts[] | undefined = useMemo(() => {
-        if (districtShapes && regionShapes) {
-            let filtered: FilteredDistricts[] = districtShapes.map(district => {
-                const scope = findScopeWithOrgUnit(scopes, district.id);
-                return {
-                    ...cloneDeep(district),
-                    region: findRegion(district, regionShapes),
-                    scope,
-                    vaccineName: scope?.vaccine,
-                };
-            }) as FilteredDistricts[];
-            if (scopes && isPolio) {
-                filtered.forEach((d, index) => {
-                    scopes.forEach(scope => {
-                        scope.group.org_units.forEach(ouId => {
-                            if (d.id === ouId) {
-                                filtered[index].vaccineName = scope.vaccine;
-                            }
-                        });
-                    });
-                });
-            }
+        if (!districtShapes || !regionShapes) return undefined;
 
-            if (searchScope) {
-                filtered = filtered.filter(d =>
-                    scopes.some(scope => scope.group.org_units.includes(d.id)),
-                );
-            }
+        const orgUnitIdToVaccine = new Map();
+        if (isPolio && scopes) {
+            scopes.forEach(scope => {
+                scope.group.org_units.forEach(ouId => {
+                    orgUnitIdToVaccine.set(ouId, scope.vaccine);
+                });
+            });
+        }
 
-            if (debouncedSearch !== '') {
-                filtered = filtered.filter(d =>
-                    d.name
+        const filtered = districtShapes.reduce((acc, district) => {
+            const scope = findScopeWithOrgUnit(scopes, district.id);
+            const vaccineName =
+                orgUnitIdToVaccine.get(district.id) || undefined;
+            const isInScope = scopes.some(sc =>
+                sc.group.org_units.includes(district.id),
+            );
+            if (
+                // Hide REJECTED or NEW org units if not already present in a scope
+                (district.validation_status !== 'VALID' && !isInScope) ||
+                (searchScope && !isInScope) ||
+                (debouncedSearch &&
+                    !district.name
                         .toLowerCase()
-                        .includes(debouncedSearch.toLowerCase()),
-                );
+                        .includes(debouncedSearch.toLowerCase()))
+            ) {
+                return acc;
             }
-            return filtered;
-        }
-        return undefined;
+
+            acc.push({
+                ...cloneDeep(district),
```

Comment:
There's now `structuredClone` that comes free with JS

https://developer.mozilla.org/en-US/docs/Web/API/structuredClone